joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -58,6 +50,22 @@ trait AstCreatorHelper { this: AstCreator =>
             nullSafeCreateParserNodeInfo(None)
   }
 
+  def cacheReferenceNode(json: Value, node: ParserNode) = {
+    // This is being called only to cache this objects
+    node match
+      case CallExpr =>
+        Try(json(ParserKeys.Fun)(ParserKeys.Obj)(ParserKeys.Decl)) match
```

Comment:
If the non-success branch is not handled, this can be a `map` or `foreach` instead, same as below

Reply:
@DavidBakerEffendi it can throw key not found error. Which we need to swallow. We had discussed this in my earlier PR.